src/randomizer.py

```diff
@@ -2048,36 +2050,18 @@ def visit_Decl(self, decl: ca.Decl) -> None:
 class Randomizer:
     def __init__(
         self,
-        settings: Optional[Mapping[str, Any]],
-        all_preset_weights: Mapping[str, Any],
+        randomization_weights: Mapping[str, float],
         rng_seed: int,
     ) -> None:
         self.random = Random(rng_seed)
 
-        custom_weights: Dict[str, int] = {}
-
-        if settings:
-            compiler_type = settings.get("compiler_type", "base")
-            custom_weights = settings["custom_weights"]
-        else:
-            compiler_type = "base"
-
-        base_weights = all_preset_weights["base"]
-        compiler_weights = all_preset_weights[compiler_type]
-
-        def create_method_weight_pair(method: Callable) -> Tuple[Callable, int]:
+        def create_method_weight_pair(
+            method: RandomizationPass,
+        ) -> Tuple[RandomizationPass, float]:
             func_name = method.__name__
+            return (method, randomization_weights[func_name])
 
-            if func_name in custom_weights:
-                weight = custom_weights[func_name]
-            elif func_name in compiler_weights:
-                weight = compiler_weights[func_name]
-            else:
-                weight = base_weights[func_name]
-
-            return (method, weight)
-
-        self.methods = list(map(create_method_weight_pair, randomization_passes))
+        self.methods = list(map(create_method_weight_pair, RANDOMIZATION_PASSES))
```

Comment:
```suggestion
        self.methods = [(method, randomization_weights[method.__name__]) for method in RANDOMIZATION_PASSES]
```
now that the inner function is a one-liner